clang/tools/clang-nvlink-wrapper/ClangNVLinkWrapper.cpp

```
@@ -165,6 +165,21 @@ void diagnosticHandler(const DiagnosticInfo &DI) {
   }
 }
 
+bool hasFatBinary(const ArgList &Args, MemoryBufferRef Buffer) {
+  if (Args.hasArg(OPT_dry_run) && Args.hasArg(OPT_assume_device_object))
+    return false;
+  if (identify_magic(Buffer.getBuffer()) != file_magic::elf_relocatable)
+    return false;
+  Expected<std::unique_ptr<ObjectFile>> ObjFile =
+      ObjectFile::createObjectFile(Buffer);
+  if (!ObjFile) {
+    (void)ObjFile.takeError();
```

This is not how you ignore errors, this will cause the program to exit due to unhandled error. You want to either `consumeError` or `errorToBool` it.

---

> This is not how you ignore errors, this will cause the program to exit due to unhandled error. You want to either `consumeError` or `errorToBool` it.

done